docs/plugins/repository-s3.asciidoc

```diff
@@ -12,98 +12,112 @@ include::install_remove.asciidoc[]
 [[repository-s3-usage]]
 ==== Getting started with AWS
 
-The plugin will default to using
+The plugin provides a repository type named `s3` which may be used when creating a repository.
+The repository defaults to using
 http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html[IAM Role]
-credentials for authentication. These can be overridden by, in increasing
-order of precedence, system properties `aws.accessKeyId` and `aws.secretKey`,
-environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_KEY`, or the
-elasticsearch config using `cloud.aws.access_key` and `cloud.aws.secret_key` or
-if you wish to set credentials specifically for s3 `cloud.aws.s3.access_key` and `cloud.aws.s3.secret_key`:
+credentials for authentication. The only mandatory setting is the bucket name:
 
-[source,yaml]
+[source,js]
 ----
-cloud:
-    aws:
-        access_key: AKVAIQBF2RECL7FJWGJQ
-        secret_key: vExyMThREXeRMm/b/LRzEB8jWwvzQeXgjqMX+6br
+PUT _snapshot/my_s3_repository
+{
+  "type": "s3",
+  "settings": {
+    "bucket": "my_bucket"
+  }
+}
 ----
+// CONSOLE
+// TEST[skip:we don't have gcs setup while testing this]
+
 
-[[repository-s3-usage-security]]
-===== Transport security
+[[repository-s3-client]]
+==== Client Settings
 
-By default this plugin uses HTTPS for all API calls to AWS endpoints. If you wish to configure HTTP you can set
-`cloud.aws.protocol` in the elasticsearch config. You can optionally override this setting per individual service
-via: `cloud.aws.ec2.protocol` or `cloud.aws.s3.protocol`.
+The client used to connect to S3 has a number of settings available. Client settings names are of
+the form `s3.client.CLIENT_NAME.SETTING_NAME` and specified inside `elasticsearch.yml`.  The
+default client name is `default`, but can be customized with the repository setting `client`.
```

Comment:
I think we should say here that people are expected to provide settings for a client named `default`.

The way I'm reading what you wrote makes me think that elasticsearch will automatically name a client: `default`.
It will actually try to find a client named `default`.

(not sure if what I say makes sense)